pkg/apply/solver/solver.go

```diff
@@ -90,23 +95,18 @@ type Options struct {
 	InventoryPolicy        inventory.InventoryPolicy
 }
 
-// Build returns the queue of tasks that have been created.
-// TODO(seans): Now that we're reporting errors, we probably
-// want to move away from the Builder patter for the TaskBuilder.
+// Build returns the queue of tasks that have been created and any validation
+// errors that occurred while building.
 func (t *TaskQueueBuilder) Build() (*TaskQueue, error) {
-	if t.err != nil {
-		return nil, t.err
-	}
-	return &TaskQueue{
-		tasks: t.tasks,
-	}, nil
+	return &TaskQueue{tasks: t.tasks}, multierror.Wrap(t.errs...)
 }
 
 // AppendInvAddTask appends an inventory add task to the task queue.
 // Returns a pointer to the Builder to chain function calls.
 func (t *TaskQueueBuilder) AppendInvAddTask(inv inventory.InventoryInfo, applyObjs object.UnstructuredSet,
 	dryRun common.DryRunStrategy) *TaskQueueBuilder {
-	klog.V(2).Infoln("adding inventory add task")
+	applyObjs = diffObjsByID(applyObjs, t.InvalidIds) // skip invalid objects
```

Comment:
Should we replace this pre-filtering with the original InvalidObjectFilter instead?

Reply:
Extracted filtering to validation.Collector